eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1

```diff
@@ -156,11 +150,24 @@ function DeployStressPackage(
     }
     $imageTag += "/$($pkg.Namespace)/$($pkg.ReleaseName):${deployId}"
 
-    $dockerFilePath = "$($pkg.Directory)/Dockerfile"
+    $dockerFilePath = if ($pkg.Dockerfile) {
+        [System.Io.Path]::Combine($pkg.Directory, $pkg.Dockerfile).ToString()
+    } else {
+        "$($pkg.Directory)/Dockerfile"
+    }
+    $dockerFilePath = [System.IO.Path]::GetFullPath($dockerFilePath)
+
     if ($pushImages -and (Test-Path $dockerFilePath)) {
         Write-Host "Building and pushing stress test docker image '$imageTag'"
         $dockerFile = Get-ChildItem $dockerFilePath
-        Run docker build -t $imageTag -f $dockerFile.FullName $dockerFile.DirectoryName
+        $dockerBuildFolder = if ($pkg.DockerBuildDir) {
+            Join-Path $pkg.Directory $pkg.DockerBuildDir
+        } else {
+            $dockerFile.DirectoryName.Trim()
+        }
+        $dockerBuildFolder = [System.IO.Path]::GetFullPath($dockerBuildFolder)
```

Comment:
If you need to do `Trim()` here then it needs to be applied to both as either input could have a trailing slash.

```suggestion
        $dockerBuildFolder = if ($pkg.DockerBuildDir) {
            Join-Path $pkg.Directory $pkg.DockerBuildDir
        } else {
            $dockerFile.DirectoryName
        }
        $dockerBuildFolder = [System.IO.Path]::GetFullPath($dockerBuildFolder).Trim()
```